### PR TITLE
New modal

### PR DESCRIPTION
**Pull request by warunicorn19**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by warunicorn19** [comment]
Aah i guess we wont be able to create new PR's on this branch unless and until the previous is merged.

**Comment by warunicorn19** [comment]
@sudo-NithishKarthik i think the modal here is okay, could we merge this if everything else seems fine? I'll do the `styled` API in a separate PR.

File: ui/components/Modal.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * }} props
+ * @returns 
+ */
+
+export const CustomModal = ({
+    open, Content, handleClose, container
+  }) =>  { const theme = useTheme();
+    return (
+      <CustomModalWrapper
+        sx={{theme}}
```
Comment (sudo-NithishKarthik) [inline ui/components/Modal.js:27]: what's the purpose of this ?

Comment (warunicorn19) [reply ui/components/Modal.js:27]: This styled api thing is confusing the hell out of me, I thought this is how the style is called🤦, but now that i use my small little brain I understand that <CustomModalWrapper /> tag would do that, right?

Comment (sudo-NithishKarthik) [reply ui/components/Modal.js:27]: yes, you actually need not use that `sx` prop here

File: ui/components/Modal.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        open={open}
+        onClose={handleClose}
+        closeAfterTransition
+        BackdropComponent={Backdrop}
+        BackdropProps={{ timeout : 200, }}
+        container={container}
+      >
+        <Fade in={open} style={{ maxHeight : "90vh", overflow : "auto" }} >{Content}</Fade>
```
Comment (sudo-NithishKarthik) [inline ui/components/Modal.js:35]: we can use `sx` instead of `style` to keep it consistent.

File: ui/components/Modal.js (hunk 1 above)
Comment (sudo-NithishKarthik) [reply ui/components/Modal.js:27]: other than that, everything else is fine...

Comment (warunicorn19) [reply ui/components/Modal.js:27]: Okay this is good?

File: ui/components/Modal.js (hunk 2 above)
Comment (warunicorn19) [reply ui/components/Modal.js:35]: Cool.

File: ui/components/Modal.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * @param {{
+ *  open?: boolean,
+ *  Content?: JSX.Element,
+ *  handleClose?: (event: {}, reason: "backdropClick" | "escapeKeyDown") => void,
+ *  container?: React.ReactInstance | (() => React.ReactInstance)
+ * }} props
+ * @returns 
+ */
+
+export const CustomModal = ({
```
Comment (sudo-NithishKarthik) [inline ui/components/Modal.js:22]: Need not export here since you are exporting it using default export at the end.

Comment (sudo-NithishKarthik) [reply ui/components/Modal.js:22]:
```suggestion
const CustomModal = ({
```

Comment (warunicorn19) [reply ui/components/Modal.js]: committed the suggestion @sudo-NithishKarthik

**Pull request merged**